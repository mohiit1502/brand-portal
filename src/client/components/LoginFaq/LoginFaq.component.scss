.c-LoginFaq {
  background: white;
  padding-top: 1.5rem;
  position: relative;

  &.scrolled:before {
    content: "";
    position: absolute;
    width: 100%;
    height: 100%;
    animation: blink 1s;
  }

  @keyframes blink {
    0% {
      background: rgba(255,255,255,0.2);
    }
    50% {
      background: rgba(255,255,255,0.8);
      // display: block;
    }
    100% {
      background: none;
      // display: none;
    }
  }

  .c-LoginFaq__header {
      padding: 0 2.5rem 1.125rem;
      border-bottom: 1px solid #CCCCCC;
  }

  .c-Accordion__button {
    padding: 1.4rem 3rem 1.4rem 2.5rem;
  }

  .c-Accordion__panel {
      padding: 0 3rem 0 2.5rem;

      &.expanded {
        padding-bottom: 1.5rem;
      }
  }
}
