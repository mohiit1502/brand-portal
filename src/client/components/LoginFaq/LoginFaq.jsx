import React from "react";
import PropTypes from "prop-types";
import FaqSingle from "./../FaqSingle";
import "./LoginFaq.component.scss";

const LoginFaq = props => {
  const {faq} = props;
  const contentRendered =  faq.items.map((item, index) => {
    item.id = `loginfaq-${index}`;
    return <FaqSingle key={index} data={item} />;
  });

  return (
    <div id={faq.id} className="c-LoginFaq">
      <div className="c-LoginFaq__header h4">{faq.header}</div>
      <div className="c-LoginFaq__content">
        {contentRendered}
      </div>
    </div>
  );
};

LoginFaq.propTypes = {
  faq: PropTypes.object
};

export default LoginFaq;