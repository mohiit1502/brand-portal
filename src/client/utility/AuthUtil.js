import authConfig from "./../config/authorizations";
import Helper from "./helper";

export default class AuthUtil {
  static isActionAccessible (action, userProfile, actedOnRole) {
    if (action && action.indexOf(".") > -1) {
      const actionConfig = Helper.search(action, authConfig);
      if (actionConfig && userProfile && userProfile.role && userProfile.role.name && typeof actionConfig === "object") {
        const role = userProfile.role.name;
        if (actionConfig.length !== undefined) {
          return actionConfig.includes(role);
        } else {
          if (Object.keys(actionConfig).includes(role) && actedOnRole && actionConfig[role].includes(actedOnRole)) {
            return true;
          }
        }
      }
    }
    return false;
  }
}
