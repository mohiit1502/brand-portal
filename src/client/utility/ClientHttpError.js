export default class ClientHttpError {

  status;
  error;
  message;

  constructor (status, error, message) {
    this.status = status;
    this.error = error;
    this.message = message;
  }

}
