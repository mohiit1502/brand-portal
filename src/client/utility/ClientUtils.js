import queryString from "query-string";

export default class ClientUtils {

  static getQueryParams (location) {
    return queryString.parse(location.search);
  }
}
