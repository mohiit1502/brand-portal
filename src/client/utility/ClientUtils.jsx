import React from "react";
import queryString from "query-string";
import {Redirect} from "react-router";

export default class ClientUtils {

  static getQueryParams (location) {
    return queryString.parse(location.search);
  }

  static  getAllValuesFromRecursiveTree(tree) {

    if (typeof tree !== "object" || tree === null) {
      return [tree];
    }

    const list = [];

    for (const i in tree) {
      if (tree.hasOwnProperty(i)) {
        list.push(...this.getAllValuesFromRecursiveTree(tree[i]));
      }
    }
    return list;
  }

  static where (arrayList, filterObj) {
    let index = -1;
    for (let i = 0; i < arrayList.length; i++) {
      for (const j in filterObj) {
        if (filterObj.hasOwnProperty(j) && arrayList[i].hasOwnProperty(j) && filterObj[j] === arrayList[i][j]) {
          index = i;
        } else {
          index = -1;
        }
      }
      if (index !== -1) {
        return index;
      }
    }
    return index;
  }

}
