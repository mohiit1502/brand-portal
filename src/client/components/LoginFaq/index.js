import LoginFaq from "./LoginFaq.jsx";

export default LoginFaq;
