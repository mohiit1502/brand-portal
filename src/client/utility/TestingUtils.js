import {applyMiddleware, compose, createStore} from "redux";
import rootReducer from "../reducers";
import thunk from "redux-thunk";
import Immutable from "immutable";
import configureMockStore from 'redux-mock-store';
import ClientHttpError from "./ClientHttpError";

export const findByTestAttribute = (component,attr) => component.find(`[data-test="${attr}"]`);

export const testStore = state => configureMockStore()(state);

export const realStore = state => {
  const configureStore = initialState => {
    const composeEnhancers =
      typeof window === "object" &&
      window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ ?
        window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__({
          serialize: { // prettier-ignore
            immutable: Immutable
          }
        }) : compose;
    const store = createStore(rootReducer, initialState, composeEnhancers(applyMiddleware(thunk)));

    if (module.hot) {
      module.hot.accept("./reducers", () => {
        const nextRootReducer = require("./reducers").default;
        store.replaceReducer(nextRootReducer);
      });
    }

    return store;
  };

  return configureStore(state || window.__WML_REDUX_INITIAL_STATE__);
}

export const clearKeys = (tree, arr) => {
  if (arr && arr.indexOf(tree) > -1) {
    return;
  }
  if (tree && typeof tree === 'object') {
    if (tree.inputId) {
      return;
    }
    arr.push(tree)
    if (tree.key) {
      try {
        delete tree.key;
      } catch (e) {}
    }
    if (Array.isArray(tree)) {
      tree.forEach(obj => clearKeys(obj, arr));
    } else {
      Object.keys(tree).forEach(key => clearKeys(tree[key], arr));
    }
  }
}

export function setupFetchStub(data) {
  return function fetchStub(_url) {
    return new Promise((resolve) => {
      resolve({
        json: () =>
          Promise.resolve({
            data,
          }),
      })
    })
  }
}

export function setupFetchThrowStub() {
  const error = new ClientHttpError(500, "test-error"); // Directly throwing new ClientHttpError fails in Jest
  throw error;
}
