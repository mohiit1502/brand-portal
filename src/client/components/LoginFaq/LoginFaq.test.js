/* eslint-disable filenames/match-regex, no-unused-vars, no-undef */
import React, {useRef} from "react";
import LoginFaq from "./LoginFaq";
import renderer from "react-test-renderer";
import faqMock from "../../../../test/client/mocks/faqMock";
import {BrowserRouter} from "react-router-dom";
import {Provider} from "react-redux";
import {testStore} from "../../utility/TestingUtils";

let store;
const setUp = (props) => {
  const mockStore = {
  };
  store = testStore(mockStore);
  return renderer.create(<Provider store={store}><BrowserRouter><LoginFaq {...props} /></BrowserRouter></Provider>);
};

jest.mock("react", () => {
  const originReact = jest.requireActual("react");
  const mUseRef = jest.fn();
  return {
    ...originReact,
    useRef: mUseRef
  };
});

describe("Login FAQ Tests", () => {

  let wrapper;
  beforeEach(() => {
    jest.restoreAllMocks();
    jest.resetAllMocks();
  });

  it("renders without error", () => {
    const mRef = {current: document.createElement("div")};
    useRef.mockReturnValue(mRef);
    wrapper = setUp(faqMock);
    const tree = wrapper.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
