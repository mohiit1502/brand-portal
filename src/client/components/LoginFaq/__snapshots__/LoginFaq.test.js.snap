// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login FAQ Tests renders without error 1`] = `
<div
  className="c-LoginFaq"
  id="LoginFAQ"
>
  <div
    className="c-LoginFaq__header h4"
  >
    Frequently Asked Questions
  </div>
  <div
    className="c-LoginFaq__content"
  >
    <div
      className="c-FaqSingle"
      id="loginfaq-0"
    >
      <div
        className="c-Accordion"
      >
        <div
          className="c-Accordion__button-container"
          data-test="accordion-buttonPanel"
        >
          <button
            className="c-Accordion__button"
            data-test="accordion-button"
            onClick={[Function]}
          >
            Why Walmart Brand Portal?
          </button>
        </div>
      </div>
    </div>
    <div
      className="c-FaqSingle"
      id="loginfaq-1"
    >
      <div
        className="c-Accordion"
      >
        <div
          className="c-Accordion__button-container"
          data-test="accordion-buttonPanel"
        >
          <button
            className="c-Accordion__button"
            data-test="accordion-button"
            onClick={[Function]}
          >
            How can I check the status of my Walmart Brand Portal application?
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
